### ISSUE TITLE
DatePicker & PasswordInput: Screen Reader doesn't announce error message in Form.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2406]
Please see this password code
https://codesandbox.io/p/devbox/react-magma-example-forked-dq6nrm
We only want to have one field on the form and be able to hear that error on the bottom of the input. No other alerts .. etc. Using VO right now, we don't hear the error

Here you will see your new datepicker input
https://codesandbox.io/p/sandbox/react-magma-example-forked-qmkmg2
With VO we don't hear any of these errors but focus returns to the fields
I also notice same with TimePicker where error is not announced but focus is moved
https://codesandbox.io/p/devbox/react-magma-example-forked-k7f7qr?workspaceId=ws_UUfRFinNFpRPK24ozBo8Lb

TimePicker has incorrect `aria-label` attributes for `hour, minute` fileds.

<img width="1478" height="954" alt="Image" src="https://github.com/user-attachments/assets/3565f2c6-08b7-4188-ba47-dfcbc4c618dd" />

`DateFiledInput` has incorrect attributes.

<img width="1097" height="478" alt="Image" src="https://github.com/user-attachments/assets/443caa8b-5954-4cb1-8360-497b76d54ec5" />

`Time Picker` NVDA and VO announces labelText when the user navigates across the input.

<img width="788" height="548" alt="Image" src="https://github.com/user-attachments/assets/bdf81b22-aebd-446b-8dad-ac25b1885452" />

[Slack](https://cengage.slack.com/archives/C0A9TSB75P0/p1774978755253869)

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--default
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--default

Everything works fine EXCEPT 1 small issue:

On macOS + Safari + VoiceOver, after clicking the `Submit` button in the `Form -> Date Picker Form` example, the focus immediately moves to the `mm` field, so the error message is skipped. Should we address this edge case? 

Everything else works perfectly.

https://github.com/user-attachments/assets/f88fe6cc-b094-46cc-ad9a-4c8d97b16f4b
